src/applications/static-pages/ics-generator.js

```diff
@@ -0,0 +1,48 @@
+import * as ICS from 'ics-js';
+
+export function icsCreate() {
+  const cal = new ICS.VCALENDAR();
+  const event = new ICS.VEVENT();
+
+  let start;
+  let end;
+  let location;
+  let description;
+
+  if (document.getElementById('starttime')) {
+    start = document.getElementById('starttime').value;
+  }
+  if (document.getElementById('endtime')) {
+    end = document.getElementById('endtime').value;
+  }
+  if (document.getElementById('location')) {
+    location = document.getElementById('location').value;
+  }
+  if (document.getElementById('description')) {
+    description = document.getElementById('description').value;
+  }
+
```

Comment:
If you follow the suggestion below with using the `data-calendar`, you could get access to the calendar data using -

```js
const addToCalendarLink = document.getElementById('add-to-calendar-link');
const fieldData = JSON.parse(addToCalendarLink.dataset.calendar);
```

[More info](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) on `data` attributes if you're interested.

Reply:
This looks good, but I think `addToCalendarLink.getAttribute('data-description');` should be the same result as `addToCalendarLink.dataset.description`. I think the difference here is just preference.

Reply:
yeah - I was having a little bit of a headache getting the latter to work, so I defaulted into the former, @ncksllvn

Reply:
Ah yeah there are some [name conversions](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset) that happen there that make it a little confusing.